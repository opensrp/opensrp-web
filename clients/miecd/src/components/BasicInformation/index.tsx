import React from 'react';
import { Card, CardTitle, Col, Row, Table } from 'reactstrap';
import { BASIC_INFORMATION } from '../../constants';
import './index.css';

export interface LabelValuePair {
    label: string;
    value: string | number;
}

interface Props {
    labelValuePairs: LabelValuePair[];
}

export default function BasicInformation({ labelValuePairs = [] }: Props) {
    return (
        <Row id={'detailsRow'}>
            <Card id={'detailsCard'}>
                <CardTitle>{BASIC_INFORMATION}</CardTitle>
                <Row>
                    <Col className={'detailsColumn'} sm="12" lg="6">
                        <Table borderless={true}>
                            <tbody>
                                {labelValuePairs.map((labelValuePair: LabelValuePair, index: number) => {
                                    if (index % 2) {
                                        return (
                                            <tr key={index}>
                                                <td>{labelValuePair.label}</td>
                                                <td>{labelValuePair.value}</td>
                                            </tr>
                                        );
                                    } else {
                                        return null;
                                    }
                                })}
                            </tbody>
                        </Table>
                    </Col>
                    <Col className={'detailsColumn'} sm="12" lg="6">
                        <Table borderless={true}>
                            <tbody>
                                {labelValuePairs.map((labelValuePair: LabelValuePair, index: number) => {
                                    if (!(index % 2)) {
                                        return (
                                            <tr key={index}>
                                                <td>{labelValuePair.label}</td>
                                                <td>{labelValuePair.value}</td>
                                            </tr>
                                        );
                                    } else {
                                        return null;
                                    }
                                })}
                            </tbody>
                        </Table>
                    </Col>
                </Row>
            </Card>
        </Row>
    );
}
