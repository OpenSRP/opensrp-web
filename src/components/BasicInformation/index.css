.dataItem {
  border: 1px solid black;
  width: 20em;
}

.detailsColumn {
  padding: 60px;
  display: grid;
  grid-template-columns: 10fr 1fr 10fr;
}

.detailsColumn:nth-child(1) {
  grid-template-columns: 4fr 1fr 10fr;
}

.detailsColumn:nth-child(2) {
  grid-template-columns: 10fr 1fr 4fr;
}

.right {
  text-align: right;
}

.left {
  text-align: left;
}

.row {
  margin: auto;
}

#detailsRow {
  width: 100%;
  padding: 30px 0;
}

#detailsCard {
  width: 100%;
}

#detailsCard .row {
  margin: 0 !important;
}
