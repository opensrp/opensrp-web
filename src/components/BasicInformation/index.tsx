import React from 'react';
import { Card, CardTitle, Col, Row } from 'reactstrap';
import {
  BASIC_INFORMATION,
  CURRENT_EDD,
  CURRENT_GRAVIDITY,
  CURRENT_PARITY,
  ID,
  LOCATION,
  PREVIOUS_PREGNANCY_RISK,
} from '../../constants';
import './index.css';

interface Props {
  id: string;
  currentGravidity: number;
  currentParity: number;
  location: string;
  currentEdd?: string;
  previousPregnancyRisk: string;
}

export default function BasicInformation(props: Props) {
  return (
    <Row id={'detailsRow'}>
      <Card id={'detailsCard'}>
        <CardTitle>{BASIC_INFORMATION}</CardTitle>
        <Row>
          <Col className={'detailsColumn'} sm="12" lg="6">
            <span className={'right'}>{ID}</span>
            <span />
            <span className={'left'}>{props.id}</span>
            <span className={'right'}>{CURRENT_GRAVIDITY}</span>
            <span />
            <span className={'left'}>{props.currentGravidity}</span>
            <span className={'right'}>{CURRENT_PARITY}</span>
            <span />
            <span className={'left'}>{props.currentParity}</span>
          </Col>
          <Col className={'detailsColumn'} sm="12" lg="6">
            <span className={'right'}>{LOCATION}</span>
            <span />
            <span className={'left'}>{props.location}</span>
            <span className={'right'}>{CURRENT_EDD}</span>
            <span />
            <span className={'left'}>{props.currentEdd}</span>
            <span className={'right'}>{PREVIOUS_PREGNANCY_RISK}</span>
            <span />
            <span className={'left'}>{props.previousPregnancyRisk}</span>
          </Col>
        </Row>
      </Card>
    </Row>
  );
}
