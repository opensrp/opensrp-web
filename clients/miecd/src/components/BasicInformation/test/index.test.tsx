import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import BasicInformation from '..';

const props = [
    { label: 'currentEdd', value: 'test edd' },
    { label: 'currentGravidity', value: 20 },
    { label: 'currentParity', value: 20 },
    { label: 'id', value: 'test' },
    { label: 'location', value: 'Test Location' },
    { label: 'previousPregnancyRisk', value: 'no risk' },
];

describe('BasicInformation', () => {
    it('must render without crashing', () => {
        shallow(<BasicInformation labelValuePairs={props} />);
    });
    it('must render without crashing when currentEdd is not supplied', () => {
        shallow(<BasicInformation labelValuePairs={props} />);
    });
    it('must render correctly', () => {
        const wrapper = mount(<BasicInformation labelValuePairs={props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
