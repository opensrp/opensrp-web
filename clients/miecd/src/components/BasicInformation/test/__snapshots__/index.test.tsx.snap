// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInformation must render correctly 1`] = `
<BasicInformation
  labelValuePairs={
    Array [
      Object {
        "label": "currentEdd",
        "value": "test edd",
      },
      Object {
        "label": "currentGravidity",
        "value": 20,
      },
      Object {
        "label": "currentParity",
        "value": 20,
      },
      Object {
        "label": "id",
        "value": "test",
      },
      Object {
        "label": "location",
        "value": "Test Location",
      },
      Object {
        "label": "previousPregnancyRisk",
        "value": "no risk",
      },
    ]
  }
>
  <Row
    id="detailsRow"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <div
      className="row"
      id="detailsRow"
    >
      <Card
        id="detailsCard"
        tag="div"
      >
        <div
          className="card"
          id="detailsCard"
        >
          <CardTitle
            tag="div"
          >
            <div
              className="card-title"
            >
              Basic Information
            </div>
          </CardTitle>
          <Row
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="row"
            >
              <Col
                className="detailsColumn"
                lg="6"
                sm="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="detailsColumn col-sm-12 col-lg-6"
                >
                  <Table
                    borderless={true}
                    responsiveTag="div"
                    tag="table"
                  >
                    <table
                      className="table table-borderless"
                    >
                      <tbody>
                        <tr
                          key="1"
                        >
                          <td>
                            currentGravidity
                          </td>
                          <td>
                            20
                          </td>
                        </tr>
                        <tr
                          key="3"
                        >
                          <td>
                            id
                          </td>
                          <td>
                            test
                          </td>
                        </tr>
                        <tr
                          key="5"
                        >
                          <td>
                            previousPregnancyRisk
                          </td>
                          <td>
                            no risk
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </Table>
                </div>
              </Col>
              <Col
                className="detailsColumn"
                lg="6"
                sm="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="detailsColumn col-sm-12 col-lg-6"
                >
                  <Table
                    borderless={true}
                    responsiveTag="div"
                    tag="table"
                  >
                    <table
                      className="table table-borderless"
                    >
                      <tbody>
                        <tr
                          key="0"
                        >
                          <td>
                            currentEdd
                          </td>
                          <td>
                            test edd
                          </td>
                        </tr>
                        <tr
                          key="2"
                        >
                          <td>
                            currentParity
                          </td>
                          <td>
                            20
                          </td>
                        </tr>
                        <tr
                          key="4"
                        >
                          <td>
                            location
                          </td>
                          <td>
                            Test Location
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </Table>
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </Card>
    </div>
  </Row>
</BasicInformation>
`;
